Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -642,27 +656,48 @@ this has no time dependence prior to this operation.
 void ComponentInfo::merge(const ComponentInfo &other) {
   if (!m_scanCounts)
     this->initScanCounts();
-  const auto toMerge = buildMergeIndices(other);
-  if (m_isSyncScan)
-    throw std::runtime_error(
-        "Sync scans not currently supported for ComponentInfo");
-  initIndices();
-  auto scanCounts(m_scanCounts);
-  for (size_t linearIndex = 0; linearIndex < other.m_positions->size();
-       ++linearIndex) {
-    if (!toMerge[linearIndex])
-      continue;
-    auto newIndex = getIndex(other.m_indices, linearIndex);
-    const size_t compIndex = newIndex.first;
-    newIndex.second += scanCount(compIndex);
-    scanCounts.access()[compIndex]++;
-    m_indexMap.access()[compIndex].push_back((*m_indices).size());
-    m_indices.access().push_back(newIndex);
-    m_positions.access().push_back((*other.m_positions)[linearIndex]);
-    m_rotations.access().push_back((*other.m_rotations)[linearIndex]);
-    m_scanIntervals.access().push_back((*other.m_scanIntervals)[linearIndex]);
-  }
-  m_scanCounts = std::move(scanCounts);
+  if (m_isSyncScan) {
+    const auto &toMerge = buildMergeIndicesSync(other);
+    for (size_t timeIndex = 0; timeIndex < other.m_scanIntervals->size();
+         ++timeIndex) {
+      if (!toMerge[timeIndex])
+        continue;
+      auto &scanIntervals = m_scanIntervals.access();
+      auto &positions = m_positions.access();
+      auto &rotations = m_rotations.access();
+      m_scanCounts.access()[0]++;
+      scanIntervals.push_back((*other.m_scanIntervals)[timeIndex]);
+      const size_t indexStart = other.linearIndex({0, timeIndex});
+      size_t indexEnd = indexStart + nonDetectorSize();
+      positions.insert(positions.end(), other.m_positions->begin() + indexStart,
+                       other.m_positions->begin() + indexEnd);
+      rotations.insert(rotations.end(), other.m_rotations->begin() + indexStart,
+                       other.m_rotations->begin() + indexEnd);
```

Comment:
I am wondering if as a first stage it would simplify things if we supported *only* sync scans in `ComponentInfo`? Async scans could still be done directly via `DetectorInfo` of course.

Reply:
If we are happy not to support async scans I'll remove the code. I think that now we have decided to prevent scanning for any components not holding detectors, then the async scan is probably irrelevant. I'll delete this code and associated functionality.

Reply:
👍 We may want to support it later, but for the time being and given the complexity maybe introducing this one step at a time is better, until we know that it works and is useful in practice?

Reply:
I'll remove it as as single commit if I can, that way it might help to reintroduce it if we find that it is needed.